main.js

```diff
@@ -0,0 +1,280 @@
+// const axios = require('axios')
+
+let countries = [];
+let APIerror = "";
+
+// delete a DOMtree, including the parent element in the parameter to this function
+
+function delDomTree(root) {
+    for (let i = 0; i < root.children.length; i++) {
+        delDomTree(root.children[i]);
+    }
+    root.remove();
+}
+
+// if string is undefined, empty or contains less than 4 characters
+// return the empty message
+
+function cleanString(string, emptyMessage) {
+
+    let out = string ?? emptyMessage;
+    if (out.includes('[') ) out = emptyMessage;
+    return out;
+
+}
+
+
+// format an array of objects with the name attribute to a comma
+// separated string with last 2 element separated by 'and' in stead of ','.
+// so results:
+// one -> one
+// one,two -> one and two
+// one,two,thrre -> one,two and three.
+
+function A2String(A) {
+    let string = '';
+    let alen = A.length;
+    for (let i = 0; i < alen; ++i) {
+        string += cleanString(A[i].name, "unspecified others")
+        switch (i) {
+            case (alen - 1):
+                break;
+            case (alen - 2):
+                string += ' and ';
+                break;
+            default:
+                string += ', ';
+                break;
+        }
+    }
```

Comment:
Werkt prima zo hoor, maar als je in het vervolg nog een keer zoiets moet doen, dan kan ik je de `.join()` aanraden:
```
return `${A.slice(0, -1).join(",")} ${A.length > 0 ? ` and ${A.slice(-1)}` : ""}`;
```

Reply:
Goeie tip, maar het zal wel een tijdje duren voor ik dat gebruik, het duurde echt 10 minuten voor ik hem helemaal door had.
En ik houd van state machines.